### PR TITLE
Fix #2172 Viewport query filter uses wrong srsName

### LINKED ISSUE #2172: Viewport query filter uses wrong srsName
Viewport option uses srsName="EPSG:900913" but then writes the polygon in EPSG:4326

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description
Added a control on leaflet map to change srs 'EPSG:4326'.
Openlayers return a srs 'EPSG:900913' with correct values.
Due to values positioned outside the worlds limits, here has been introduced a function to split in two multipolygon the bound and extent on the Viewport spatial filter.
screenshot from 2017-09-19 12-20-04

## Requirements
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
 - [x] Bugfix

**What is the current behavior?** 
issue #2172

**What is the new behavior?**
- Added a control on leaflet map to change srs 'EPSG:4326' (Viewport)
- split bound in two polygon when view contains 180 -180 values (Viewport)

**Does this PR introduce a breaking change?** (check one with "x")
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/epics/wfsquery.js [hunk 1]
```diff
@@ -261,21 +263,11 @@ const viewportSelectedEpic = (action$, store) =>
             if (action.type === SELECT_VIEWPORT_SPATIAL_METHOD ||
                 action.type === CHANGE_MAP_VIEW && spatialFieldMethodSelector(store.getState()) === "Viewport") {
 
+                const projection = isLeaflet(store.getState()) ? 'EPSG:4326' : map.projection;
```
Comment (mbarto) [inline web/client/epics/wfsquery.js]: bounds should also contain its srs, if it doesn't we should add it and not have knowledge of the mapping lib in use

File: web/client/utils/CoordinatesUtils.js [hunk 2]
```diff
@@ -56,6 +57,100 @@ let crsLabels = {
     "EPSG:3857": "EPSG:3857"
 };
 
+const normalizePoint = (point) => {
```
Comment (mbarto) [inline web/client/utils/CoordinatesUtils.js]: You moved this, did you change anything?

File: web/client/utils/CoordinatesUtils.js [hunk 3]
```diff
@@ -56,6 +57,100 @@ let crsLabels = {
...
+    return {
+        x: point.x || 0.0,
+        y: point.y || 0.0,
+        srs: point.srs || 'EPSG:4326'
+    };
+};
+
+const reproject = (point, source, dest, normalize = true) => {
```
Comment (mbarto) [inline web/client/utils/CoordinatesUtils.js]: You moved this, did you change anything?

File: web/client/utils/CoordinatesUtils.js (hunk 2 above)
Comment (allyoucanmap) [reply web/client/utils/CoordinatesUtils.js]: no, i didn't change the function

File: web/client/utils/CoordinatesUtils.js (hunk 3 above)
Comment (allyoucanmap) [reply web/client/utils/CoordinatesUtils.js]: no, i didn't change the function

**Pull request merged**